CMakeLists.txt

```diff
@@ -143,6 +143,11 @@ target_compile_options(CLI11_warnings
 add_library(CLI11 INTERFACE)
 add_library(CLI11::CLI11 ALIAS CLI11) # for add_subdirectory calls
 
+if (CMAKE_CXX_STANDARD GREATER 16)
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 8)
```

Comment:
Why is this "not" version equal 8? This only needs to be added for C++17 & GCC < 9.

Reply:
I think it only really applies to GCC 8.  GCC 7 doesn't have <filesystem> and GCC 9 doesn't have a problem, so this was a the condition that needed the extra linking target.

Reply:
But shouldn't it be `if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 8)`?

Reply:
Yes, so why is then working on gcc 8?   I am trying with the in code check, if that works we can probably remove it from CMAKE.

Reply:
The version equal bothers me; `VERSION_LESS 9` would be better. I assume this isn't expecting 8.0.0?